src/ukv/ukv.hpp

```diff
@@ -0,0 +1,411 @@
+  write.values = value_.member_ptr();
+  ukv_write(&write);
+  return {size_t(status),
+          status ? operation_status_t::ok_k : operation_status_t::error_k};
+}
+
+operation_result_t ukv_t::update(key_t key, value_spanc_t value) {
+  status_t status;
+  ukv_key_t key_ = key;
+  ukv_byte_t *value_ = nullptr;
+
+  ukv_read_t read{};
+  read.db = db_;
+  read.error = status.member_ptr();
+  read.arena = arena.member_ptr();
+  read.options = option;
+  read.tasks_count = 1;
+  read.collections = &collection_;
+  read.keys = &key_;
+  read.values = &value_;
+  ukv_read(&read);
+  if (!status)
+    return {0, operation_status_t::not_found_k};
+
+  return upsert(key, value);
+}
+
+operation_result_t ukv_t::remove(key_t key) {
+  status_t status;
+  ukv_key_t key_ = key;
+
+  ukv_write_t write{};
+  write.db = db_;
+  write.error = status.member_ptr();
+  write.arena = arena.member_ptr();
+  write.options = option;
+  write.tasks_count = 1;
+  write.collections = &collection_;
+  write.keys = &key_;
+  ukv_write(&write);
+
+  return {status ? size_t(1) : 0,
+          status ? operation_status_t::ok_k : operation_status_t::error_k};
+}
+
+operation_result_t ukv_t::read(key_t key, value_span_t value) const {
+  status_t status;
+  ukv_key_t key_ = key;
+  ukv_byte_t *value_ = nullptr;
+  ukv_length_t *lengths = nullptr;
+
+  ukv_read_t read{};
+  read.db = db_;
+  read.error = status.member_ptr();
+  read.arena = arena.member_ptr();
+  read.options = option;
+  read.tasks_count = 1;
+  read.collections = &collection_;
+  read.keys = &key_;
+  read.lengths = &lengths;
+  read.values = &value_;
+  ukv_read(&read);
+  if (!status)
+    return {0, operation_status_t::error_k};
+
+  if (lengths[0] == ukv_length_missing_k)
+    return {0, operation_status_t::not_found_k};
+  memcpy(value.data(), value_, lengths[0]);
+  return {1, operation_status_t::ok_k};
+}
+
+operation_result_t ukv_t::batch_upsert(keys_spanc_t keys, values_spanc_t values,
+                                       value_lengths_spanc_t sizes) {
+  status_t status;
+  std::vector<ukv_length_t> offsets;
+  offsets.reserve(sizes.size() + 1);
+  offsets.push_back(0);
+  size_t offset = 0;
+  for (auto size : sizes) {
+    offset += size;
+    offsets.push_back(offset);
+  }
+
+  auto val = make_value(values.data(), values.size());
+  ukv_write_t write{};
+  write.db = db_;
+  write.error = status.member_ptr();
+  write.arena = arena.member_ptr();
+  write.options = option;
+  write.tasks_count = keys.size();
+  write.collections = &collection_;
+  write.keys = reinterpret_cast<ukv_key_t const *>(keys.data());
+  write.keys_stride = sizeof(ukv_key_t);
+  write.offsets = offsets.data();
+  write.offsets_stride = sizeof(ukv_length_t);
+  write.lengths = reinterpret_cast<ukv_length_t const *>(sizes.data());
+  write.lengths_stride = sizeof(ukv_length_t);
+  write.values = val.member_ptr();
+  ukv_write(&write);
+
+  return {status ? keys.size() : 0,
```

Comment:
A cleaner way is to put the ternary operator outside of the initialization list.